### PR TITLE
Add dtypes to DataFrame.

### LINKED ISSUE #52: DataFrame.dtypes
Raising this to discuss what the handling of the `.dtypes` should be.
Current implementations:
 - Pandas: Returns a pandas Series with index of fieldnames and values as dtypes
 - Spark: Returns a list of tuples with (fieldname, dtype)
 - Koala: Same as Spark

Other Differences:
 - Whether to show indexes in .dtypes:
   - Pandas: Shows only columns and not indexes
   - Spark: Shows all columns, no index concept
   - Koala: Shows columns and indexes
 - Type values:
   - Pandas: Shows strings, arrays, map, etc as `object`
   - Spark: Shows strings, arrays, map, etc in appropriate simpleString notation
   - Koala: Same as spark
 - Dtype notation:
   - Pandas: Shows numpy type for the dtype
   - Spark: Shows simpleString notation
   - Koala: Same as spark

### PR DESCRIPTION
**Pull request by ueshin**
Closes #52.

### CODE REVIEW

**Comment by ueshin** [comment]
cc @thunterdb

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -360,6 +360,19 @@ def columns(self, names):
...
+    @property
+    def dtypes(self):
+        """Return the dtypes in the DataFrame.
+
+        This returns a Series with the data type of each column. The result's index is the original
+        DataFrame's columns. Columns with mixed types are stored with the object dtype.
+
+        :return: :class:`pd.Series` The data type of each column.
+        """
+        return pd.Series([to_arrow_type(self._sdf.schema[col].dataType).to_pandas_dtype()
```
Comment (thunterdb) [inline databricks/koalas/frame.py]: is there any reason for not calling directly the `.dtype` of the Series? You just added the same logic down there in `series.py`.

File: databricks/koalas/tests/test_dataframe.py [hunk 2]
```diff
@@ -282,6 +282,11 @@ def test_dropna(self):
         with self.assertRaisesRegex(NotImplementedError, msg):
             ddf.dropna(axis='foo')
 
+    def test_dtype(self):
+        d = self.df
+        full = self.full
```
Comment (thunterdb) [inline databricks/koalas/tests/test_dataframe.py]: actually, you are only testing a single type here. Can you make a small dataframe with a bunch of different types in it? The pandas test suites have a few example. No need to support all the complex types, at least the basic scalar types.

File: databricks/koalas/frame.py [hunk 3]
```diff
@@ -360,6 +360,18 @@ def columns(self, names):
...
         self._metadata = self._metadata.copy(column_fields=names)
 
+    @property
+    def dtypes(self):
+        """Return the dtypes in the DataFrame.
+
+        This returns a Series with the data type of each column. The result's index is the original
+        DataFrame's columns. Columns with mixed types are stored with the object dtype.
+
+        :return: :class:`pd.Series` The data type of each column.
```
Comment (rxin) [inline databricks/koalas/frame.py]: can we add an example?

File: databricks/koalas/series.py [hunk 4]
```diff
@@ -161,8 +162,11 @@ def __sub__(self, other):
 
     @property
     def dtype(self):
-        from databricks.koalas.typing import as_python_type
-        return as_python_type(self.schema.fields[-1].dataType)
+        """Return the dtype object of the underlying data."""
```
Comment (rxin) [inline databricks/koalas/series.py]: add an example if possible

**Comment by thunterdb** [comment]
@ueshin it looks like there are doc conflicts now.

**Comment by ueshin** [comment]
@thunterdb I will fix and add examples soon.

**Comment by thunterdb** [comment]
@ueshin excellent, thank you.

**Comment by sadikovi** [comment]
Has this patch made it into the latest release? I am still experiencing "'DataFrame' object has no attribute 'dtypes'" when calling `df.dtypes` on koalas DataFrame.

**Comment by HyukjinKwon** [comment]
AFAIK another release is coming very shortly.

**Pull request merged**